[Rachael-Chan] iP

Rachael-Chan opened the pull request:
### Spiderman

> "With great power, comes great responsibility." - Uncle Ben

Spiderman helps you capture every task in your web of productivity, so you never have to remember everything on your own. Just like Spidey always stays on top of his game, you will too!

It's:
- **Text-Based**: Simple and direct, like Spidey's witty one-liners.
- **Easy to Learn**: As intuitive as Peter Parker's Spidey Sense.
- **SUPER FAST** to Use: Faster than SpiderMan swinging through the city!

All you need to do is,

1. Download it from [here](https://github.com/Rachael-Chan/ip/releases/tag/v0.2).
2. Double-click it
3. Add your tasks
4. Let it manage your tasks for you
And it's **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
          new Spiderman("tasks.txt").run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Rachael-Chan committed:
Renamed Duke to Spiderman

Rachael-Chan committed:
Added greeting and goodbye message

Rachael-Chan committed:
feat: Add support for echo

Rachael-Chan committed:
feat: Add support for adding lists

Rachael-Chan committed:
style: Removed unnecessary type argument

Rachael-Chan committed:
feat: Tasks can be marked as done and undone

Rachael-Chan committed:
feat: Added deadline, event and todo

Rachael-Chan committed:
fix: todo task not correctly displayed in list

Rachael-Chan committed:
test: Added text ui testing

Rachael-Chan committed:
feat: Added error handling

Rachael-Chan committed:
feat: Added deletion of tasks

Rachael-Chan committed:
test: Updated expected test to match changes in output

Rachael-Chan committed:
feat: Tasks added to hard disk

Rachael-Chan committed:
Merge pull request #1 from Rachael-Chan/branch-Level-7

Merge branch-Level-7 to master

Rachael-Chan committed:
feat: Tasks support datetime

Rachael-Chan committed:
Merge pull request #2 from Rachael-Chan/branch-Level-8

Merge branch-Level-8 to master

jinnsuke commented on src/main/java/Spiderman.java:
```diff
@@ -0,0 +1,212 @@
+                System.out.println(taskList.get(number).toString());
+                continue;
+            }
+
```
The code for parsing and processing user commands is quite functional, but it could benefit from refactoring to improve readability and maintainability. The if-else blocks for different commands such as delete, mark, unmark, deadline, event, and todo could be extracted into separate methods or a command handler class. This would not only streamline the main method but also make it easier to manage and extend in the future!

jinnsuke commented on src/main/java/Spiderman.java:
```diff
@@ -0,0 +1,212 @@
+                }
+
+                // If task is a deadline
+                if (savedTasks[0].equals("D")) {
```
In the loadTasksFromFile method, the if statements for checking task types (e.g., if (savedTasks[0].equals("T"))) could be refactored to improve clarity and reduce redundancy. Consider using a switch statement based on savedTasks[0] to handle different task types. This approach will make the code more readable and easier to manage as more task types are added in the future!

jinnsuke commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,38 @@
+    public String toString() {
+        return "[" + this.getStatusIcon() + "] " + this.description;
+    }
+}
```
Would be better to add some method comments!

jinnsuke reviewed (commented):
Code looks incomplete but good job so far! Jiayou!

Rachael-Chan committed:
test: Updated test to support package

Rachael-Chan committed:
feat: Added package spiderman

Rachael-Chan committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Rachael-Chan committed:
refactor: Modified build.gradle

Rachael-Chan committed:
Modified build.gradle

Rachael-Chan committed:
test: Added JUnit tests for Deadline and Todo Class

Rachael-Chan committed:
docs: Added JavaDoc comments.

Rachael-Chan committed:
Updated code to follow coding convention

Rachael-Chan committed:
feat: Added find function

Rachael-Chan committed:
Merge pull request #3 from Rachael-Chan/branch-A-JavaDoc

Merge branch-A-JavaDoc to master

Rachael-Chan committed:
Merge pull request #4 from Rachael-Chan/branch-A-CodingStandard

Merge branch-A-CodingStandard to master

Rachael-Chan committed:
Merge pull request #5 from Rachael-Chan/branch-Level-9

Merge branch-Level-9 to master

notnotmax commented on src/main/java/spiderman/Deadline.java:
```diff
@@ -0,0 +1,39 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
```
Perhaps a Javadoc header comment could be added?

notnotmax commented on src/test/java/spiderman/TodoTest.java:
```diff
@@ -0,0 +1,28 @@
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
```
I think explicit imports would make the dependencies clearer - maybe this was automatically done by the IDE?

notnotmax commented on src/main/java/spiderman/Spiderman.java:
```diff
@@ -0,0 +1,238 @@
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+import java.util.ArrayList;
```
I think it may be clearer for import statements to be ordered consistently, for example ordering imports from the same package lexicographically (ArrayList should be before Scanner). Supposedly, IDEs can be configured somehow to help you do this, or if you have caught up to week 4, see A-CheckStyle.

notnotmax commented on src/test/java/spiderman/DeadlineTest.java:
```diff
@@ -0,0 +1,35 @@
+package spiderman;
+
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
```
I think static imports could be before other non-static import statements (in accordance with the coding standard).

notnotmax reviewed (commented):
lgtm - overall, well done! There were no egregious violations of the coding standard, just a few minor errors.

Rachael-Chan committed:
refactor: Refactored spiderman with OOP

Rachael-Chan committed:
feat: Added GUI for spiderman chatbot

Rachael-Chan committed:
refactor: Modified Main.java

Removed unused method from Main class and also removed unused imports.

Rachael-Chan committed:
Merge pull request #6 from Rachael-Chan/branch-Level-10

Merge branch-Level-10 to master

Rachael-Chan committed:
feat: Added more exception handling

Updated TaskList class to handle out of bounds error when user gives a list number that is not valid.

Rachael-Chan committed:
feat: Added more exception handling

Exception handling is further added to user inputs to ensure that user does not provide commands that require integer with value other than integer.

Rachael-Chan committed:
feat: Added assertions

Assert statements were added to Command Class to ensure that these assumptions hold when the program is executed.

Rachael-Chan committed:
docs: Added JavaDoc comments

Added javadoc comments to each class in the spiderman package.
This is to update users with a clear knowledge of what each class does.

Rachael-Chan committed:
Merge pull request #7 from Rachael-Chan/branch-A-Assertions

Merge branch-A-Assertions to master

Rachael-Chan committed:
Merge remote-tracking branch 'origin' into branch-A-CodeQuality

Resolve merge conflicts

Rachael-Chan committed:
Merge pull request #8 from Rachael-Chan/branch-A-CodeQuality

Merge branch-A-CodeQuality to master

Rachael-Chan committed:
refactor: Formatted code in command class

Rachael-Chan committed:
fix: Error handling for delete, mark and unmark commands with empty inputs.

Rachael-Chan committed:
feat: Tasks can now be updated

Rachael-Chan committed:
fix: Set chatbot title for GUI

Rachael-Chan committed:
Added screenshot of chatbot GUI

Rachael-Chan committed:
fix: Changed Ui screenshot

Rachael-Chan committed:
docs: Updated user guide for chatbot

Rachael-Chan committed:
fix: Added exception handling

Added exception handling for empty
parameters when adding deadline and
event tasks.

Rachael-Chan committed:
docs: Updated user guide documentation

Rachael-Chan committed:
docs: Updated user guide documentation with table of contents

Rachael-Chan committed:
fix: Handle error message output for wrong event command format
